jsonschema-validation.xml

```diff
@@ -1017,13 +1017,16 @@
                 <t>
                     If the instance is a string, and if "contentMediaType" is present, this
                     property contains a schema which describes the structure of the string.
+                    The string MUST be in a format compatible with the data model defined in the
```

Comment:
There's not really a MUST here, because these are annotations, not assertions.  They are telling the application that it can use the data in some way.  That does not guarantee that the data is suitable.

Just because a piece of JSON data is annotated with `title` and `description` does not mean that the given title and description are accurate, relevant, or even safe to display to end-users.  It is up to the application to decide whether additional validation is needed, and what that would look like.

`contentEncoding`, `contentMediaType`, and `contentSchema` should be treated the same as if received a MIME attachment with MIME header `Content-Transfer-Encoding` matching `contentEncoding` and MIME header `Content-Type` matching `contentMediaType`.  And separately, you are given a JSON Schema, and told that it is relevant to that attached resource.

There is no automatic application of the schema.  You can decode and parse the attachment or not.  If you do, you can apply the schema or not.  You have to decide how to parse the attachment, based on its media type, into the JSON Schema content model.

It doesn't make sense to say "a format compatible with the data model" because we do not define mappings from media types into the data model.  Applications are required to come up with that on their own.  Some (like the `application/jwt` example in the spec) are fairly obvious.  Others area not, and may potentially work in several different ways that could be suitable for different use cases.

JSON Schema doesn't enforce any of that.  The application first of all has to decide whether to trust the instance author (if you have `"contentMediaType": "application/javascript"`, there is a lot of security risk to `eval()`ing it, for example).

These fields are also like HTTP headers- `Content-Type` is not guaranteed to be accurate by HTTP.  The sender may or may not have set it correctly, and some user-agents will do a bit of parsing to try to figure out the "real" media type.

But all of this means that these are not MUST assertions.  A MUST is something that an implementation has to take action on, and there is none of that with these keywords.  As noted in the introductory material for this section, `contentSchema` cannot affect the outcome of the schema that contains it.  If run, it MUST be done as a separate step (more accurately, the instance string MUST NOT be ***automatically*** decoded, parsed, and validated.

Reply:
> A MUST is something that an implementation has to take action on

I wouldn't interpret a MUST here as "the implementation must verify that this is a valid schema"\*, but rather "you can assume that this is a valid schema, and if anything takes action on this and it *isn't* a valid schema, well, we already told you it must be one, so too bad, you get to keep both halves".

\* After all, there are other places in the spec that assert a "MUST" that the implementation doesn't necessarily check.  e.g. in an if/then/else assertion, if the "if" subschema evaluates true, we may never bother to actually verify that the subschema underneath the "else" path actually is a spec-compliant schema, because we never go down that path.  To require the implementation to verify every MUST assertion in the spec incurs an unnecessary efficiency burden.

Reply:
@handrews my understanding from your comment (https://github.com/json-schema-org/json-schema-spec/pull/917#discussion_r428091385) is that there's a schema to be applied to some data model. I get your point on the MUST. Maybe changing to a SHOULD would suffice? Pushing now...

Reply:
@handrews My logic: On one hand, you've said that the whole point of there being a schema (https://github.com/json-schema-org/json-schema-spec/pull/917#discussion_r428091385) is to apply it to some data model that the instance uses, but on the other you're saying it can't be a MUST, which makes sense because these aren't assertions. I've changed to SHOULD because why else would there be a schema in the first place? Resolving this conversation because I just force pushed the change.